File: cache/src/main/java/com/osrsd/cache/def/ItemDefinition.java
package com.osrsd.cache.def;

import lombok.Data;

import java.util.HashMap;
import java.util.Map;

@Data
public final class ItemDefinition implements Definition {

    private int id;
    private String name;
    private int resizeX;
    private int resizeY;
    private int resizeZ;
    private int xan2d;
    private int yan2d;
    private int zan2d;
    private int cost;
    private boolean isTradeable;
    private int stackable;
    private int inventoryModel;
    private boolean members;
    private short[] colorFind;
    private short[] colorReplace;
    private short[] textureFind;
    private short[] textureReplace;
    private int zoom2d;
    private int xOffset2d;
    private int yOffset2d;
    private int ambient;
    private int contrast;
    private int[] countCo;
    private int[] countObj;
    private String[] options;
    private String[] interfaceOptions;
    private int maleModel0;
    private int maleModel1;
    private int maleModel2;
    private int maleOffset;
    private int maleHeadModel;
    private int maleHeadModel2;
    private int femaleModel0;
    private int femaleModel1;
    private int femaleModel2;
    private int femaleOffset;
    private int femaleHeadModel;
    private int femaleHeadModel2;
    private int notedID;
    private int notedTemplate;
    private int team;
    private int shiftClickDropIndex;
    private int boughtId;
    private int boughtTemplateId;
    private int placeholderId;
    private int placeholderTemplateId;
    private Map<Integer, Object> params;

    public ItemDefinition(int id) {
        this.id = id;
        name = "null";
        resizeX = 128;
        resizeY = 128;
        resizeZ = 128;
        xan2d = 0;
        yan2d = 0;
        zan2d = 0;
        cost = 1;
        stackable = 0;
        members = false;
        zoom2d = 2000;
        xOffset2d = 0;
        yOffset2d = 0;
        options = new String[]{null, null, "Take", null, null};
        interfaceOptions = new String[]{null, null, null, null, "Drop"};
        maleModel0 = -1;
        maleModel1 = -1;
        maleModel2 = -1;
        maleHeadModel = -1;
        maleHeadModel2 = -1;
        femaleModel0 = -1;
        femaleModel1 = -1;
        femaleModel2 = -1;
        femaleHeadModel = -1;
        femaleHeadModel2 = -1;
        notedID = -1;
        notedTemplate = -1;
        shiftClickDropIndex = -2;
        boughtId = -1;
        boughtTemplateId = -1;
        placeholderId = -1;
        placeholderTemplateId = -1;
        params = new HashMap<>();
    }

}

File: app/src/main/kotlin/com/osedit/spring/domain/ItemDefinition.kt
package com.osedit.spring.domain

import javax.persistence.ElementCollection
import javax.persistence.Entity
import javax.persistence.Id

@Entity
data class ItemDefinition(@Id val id: Int? = 0) : Definition {
    var name: String = "null"
    var resizeX: Int = 128
    var resizeY: Int = 128
    var resizeZ: Int = 128
    var xan2d: Int = 0
    var yan2d: Int = 0
    var zan2d: Int = 0
    var cost: Int = 1
    var isTradeable: Boolean = false
    var stackable: Int = 0
    var inventoryModel: Int = 0
    var members: Boolean = false
    var colorFind: ShortArray? = null
    var colorReplace: ShortArray? = null
    var textureFind: ShortArray? = null
    var textureReplace: ShortArray? = null
    var zoom2d: Int = 2000
    var xOffset2d: Int = 0
    var yOffset2d: Int = 0
    var ambient: Int = 0
    var contrast: Int = 0
    var countCo: IntArray? = null
    var countObj: IntArray? = null
    @ElementCollection
    var options = mutableListOf(null, null, "Take", null, "Drop")
    @ElementCollection
    var interfaceOptions = mutableListOf(null, null, null, null, "Drop")
    var maleModel0: Int = -1
    var maleModel1: Int = -1
    var maleModel2: Int = -1
    var maleOffset: Int = 0
    var maleHeadModel: Int = -1
    var maleHeadModel2: Int = -1
    var femaleModel0: Int = -1
    var femaleModel1: Int = -1
    var femaleModel2: Int = -1
    var femaleOffset: Int = -1
    var femaleHeadModel: Int = -1
    var femaleHeadModel2: Int = -1
    var notedID: Int = -1
    var notedTemplate: Int = -1
    var team: Int = 0
    var shiftClickDropIndex: Int = -2
    var boughtId: Int = -1
    var boughtTemplateId: Int = -1
    var placeholderId: Int = -1
    var placeholderTemplateId: Int = -1
    @ElementCollection
    var params = mutableMapOf<Int, String>()
}